### ISSUE TITLE
Key Vault resource name contains a trademarked or reserved word

### ISSUE THREAD

**Issue by charlesakalugwu** [openshift/openshift-azure#1662]
Caught a flake with the following error signature:

```
time="2019-05-16T16:19:44Z" level=debug msg="400 Bad Request: Failed to apply request: keyvault.VaultsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code=\"ReservedResourceName\" Message=\"The resource name 'kv-oxkymednnqtltmfonxbox' or a part of the name is a trademarked or reserved word.\"" func="github.com/openshift/openshift-azure/pkg/fakerp.(*Server).badRequest" file="/tmp/tmp.FYboeSjNrK/src/github.com/openshift/openshift-azure/pkg/fakerp/util.go:11"
```

The keyvault resource name contains `xbox` which is a trademarked word :man_shrugging: 

/kind test-flake

**Comment by charlesakalugwu** [comment]
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/1660/pull-ci-azure-master-upgrade-v4.4/33/build-log.txt

**Comment by ehashman** [comment]
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-reserved-resource-name

**Comment by ehashman** [comment]
See also https://github.com/openshift/openshift-azure/pull/1595

**Comment by ehashman** [comment]
/assign

I'll see if I can figure out why @jim-minter's PR didn't fix this.

**Comment by ehashman** [comment]
Here was @jim-minter's error from April 23rd. The key-vault was also affected:

```
FATA[2019-04-23T16:31:20-05:00] containerservice.OpenShiftManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: error response cannot be parsed: "400 Bad Request: Failed to apply request: keyvault.VaultsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code=\"ReservedResourceName\" Message=\"The resource name 'kv-wfkxboxhepbqjjfsjxond' or a part of the name is a trademarked or reserved word.\"\n" error: json: cannot unmarshal number into Go value of type azure.RequestError 
```

So the key vault name gets set here: https://github.com/openshift/openshift-azure/blob/e6d7fece70d7f0da77c39dc31c2f6fc4bc5a55e2/pkg/fakerp/vault.go#L156

It comes directly from the `vaultURL` that appears to be set in the FakeRP: https://github.com/openshift/openshift-azure/blob/e6d7fece70d7f0da77c39dc31c2f6fc4bc5a55e2/pkg/fakerp/fakerp.go#L188-L194

I'm guessing to fix this we will need to validate that the keyVaultURL doesn't contain any reserved words. Right now we don't do that: https://github.com/openshift/openshift-azure/blob/e6d7fece70d7f0da77c39dc31c2f6fc4bc5a55e2/pkg/api/validate/types.go#L48

So this is why the fix from #1595 doesn't address this; validation isn't catching the real issue and that code path isn't the one being exercised in generating this random name.

**Comment by ehashman** [comment]
Oh, aha, found where the vaultURL is being generated: https://github.com/openshift/openshift-azure/blob/e6d7fece70d7f0da77c39dc31c2f6fc4bc5a55e2/pkg/fakerp/fakerp.go#L279

Looks like we missed checking if the vaultURL has Microsoft reserved keywords... patch incoming: https://github.com/openshift/openshift-azure/blob/e6d7fece70d7f0da77c39dc31c2f6fc4bc5a55e2/pkg/util/random/random.go#L74-L81
